segtypes/common/codesubsegment.py

```diff
@@ -273,6 +273,32 @@ def determine_symbols(self):
                 while len(possible_jtbl_jumps) > 0 and possible_jtbl_jumps[0][0] < hi_insn.address:
                     del possible_jtbl_jumps[0]
 
+                # Find gp relative reads and writes e.g  lw $a1, 0x670($gp)
+                if hi_insn.op_str.endswith("($gp)"):
+                    gp_base = options.get_gp()
+                    if gp_base is None:
+                        print("gp_value not set in yaml, can't calculate %gp_rel reloc value")
+                    else:
+                        op_split = hi_insn.op_str.split(", ")
+                        gp_offset = op_split[1][:-5] # extract the 0x670 part
+                        gp_offset = int(gp_offset, 16)
+                        symbol_addr = gp_base + gp_offset
+                        
+                        sym = self.parent.create_symbol(symbol_addr, offsets=True, reference=True)
+                        offset = symbol_addr - sym.vram_start
+                        if offset != 0:
+                            offset_str = f"+0x{offset:X}"
+
+                        if self.parent:
+                            self.parent.check_rodata_sym(func_addr, sym)
+
+                        self.update_access_mnemonic(sym, hi_insn.mnemonic)
+
+                        func.insns[i].is_gp = True
+                        func.insns[i].hi_lo_sym = sym
+                        func.insns[i].sym_offset_str = offset_str
+                        continue
```

Comment:
I could be wrong, but I'm concerned that `continue` here is skipping important stuff below that needs to happen in general...maybe not true for this case though